js/modules/autoOrientImage.js

```diff
@@ -0,0 +1,29 @@
+const loadImage = require('blueimp-load-image');
+
+// File | Blob | URLString -> LoadImageOptions -> Promise<DataURLString>
+//
+// Documentation for `options`:
+// https://github.com/blueimp/JavaScript-Load-Image/tree/v2.18.0#options
+exports.autoOrientImage = (fileOrBlobOrURL, options) => {
+  const optionsWithAutoOrient = Object.assign(
+    {},
+    options, {
+      canvas: true,
+      orientation: true,
+    }
+  );
+
+  return new Promise((resolve, reject) => {
+    loadImage(fileOrBlobOrURL, canvasOrError => {
+      if (canvasOrError.type === 'error') {
+        const error = canvasOrError;
+        reject(error);
+        return;
+      }
+
+      const canvas = canvasOrError;
+      const dataURL = canvas.toDataURL();
+      resolve(dataURL);
+    }, optionsWithAutoOrient);
+  });
+};
```

Comment:
Started this `modules` directory where we can write new code as CommonJS modules. This will allow us to write smaller modules that are testable on their own, potentially even without `window` if it’s not required for a given module.

Let me know what you think 😄

Reply:
I'd like to avoid upper case in our filenames, maybe use snake or kebab case. I've run into too many situations where it worked on Windows/OSX but not Linux...

Reply:
I hear you. I’ve run into this before as well but I learned it’s easy to prevent using a lint rule such as `eslint-plugin-require-path-exists`. It would give you an error as follows:
```
72:39  error    Casing of ./js/modules/autoorientimage does not match the underlying filesystem  import/no-unresolved
```

I am happy to change it to `snake_case` for now for consistency but down the line when we use tools such as React, TypeScript, or PureScript, it’s fairly standard to name files after the component / module they represent, e.g. see this discussion: https://github.com/felixge/node-style-guide/issues/30#issuecomment-61816649

Reply:
Fun historical fact! I submitted the PR that allows `eslint-plugin-filenames` to transform your `camelCase` exported variable to a different-format filename: https://github.com/selaux/eslint-plugin-filenames/pull/9

Reply:
Awesome!